lib/common/client/base.rb

```diff
@@ -23,7 +26,17 @@ def config
 
       # memoize the connection from config
       def connection
-        @connection ||= config.connection
+        @connection ||= lambda do
+          connection = config.connection
+          handlers = connection.builder.handlers
+
+          if handlers.include?(Faraday::Adapter::HTTPClient) &&
```

Comment:
I like that this protection is on by default, and prevents developer error attempting to use an adapter with a known issue. 

In an ideal world I'd like to see an adapter specific flag for this coming from faraday: `connection.cookies_enabled?`, but I bet we'd have trouble recommending this kind of contribution.